runtime/interp.c

```diff
@@ -722,9 +788,11 @@ value caml_interprete(code_t prog, asize_t prog_size)
       mlsize_t i;
       value block;
       if (size <= Max_young_wosize / Double_wosize) {
-        Alloc_small(block, size * Double_wosize, Double_array_tag);
+        Alloc_small(block, size * Double_wosize, Double_array_tag, Enter_gc);
       } else {
+        Setup_for_gc;
```

Comment:
This change looks dubious. If the calling convention of `caml_alloc_shr` has actually changed, it should be documented and the other calls should be protected accordingly.

Reply:
I agree with @silene 's remark.

Reply:
Fixed in https://github.com/ocaml/ocaml/pull/10831/commits/e7a8ac1b8f95c7cd6dc837e1d37cfba0688b3b84.